src/aia_clic_extension.adoc

```diff
@@ -0,0 +1,1816 @@
+`clicintip[__i__]` behavior is unaffected by `clicintie[__i__]` setting.
+
+
+The value in the `clicintip[__i__]` is undefined when switching from
+level-sensitive mode to edge-triggered mode in `clicintattr[__i__]`.
+
+NOTE: Software cannot rely on the underlying `clicintip[__i__]`
+register bits used in edge-triggered mode to hold state while in
+level-sensitive mode.
+
+=== CLIC Interrupt Enable (`clicintie`)
+Each interrupt has an associated interrupt enable.
+For the first 64 interrupts, these bits mirror the values of mie CSR.
+
+Each interrupt input has a dedicated interrupt-enable WARL bit (`clicintie[__i__]`)
+This control bit is read-write to enable/disable the corresponding interrupt.
+Software should assume `clicintie[__i__]`=0 means no interrupt enabled, and `clicintie[__i__]`=1 indicates an interrupt is enabled.
+
+NOTE: `clicintie[__i__]` is the individual enable bit while {status}.{ie} is
+the global enable bit for the current privilege mode. Therefore, for an
+interrupt `_i_` to be enabled in the current privilege mode, both `clicintie[__i__]`
+and {status}.{ie} have to be set.
+
+
+NOTE: In contrast, since {status}.{ie} only takes effect in the current privilege
+mode according to RISC-V convention, an interrupt `_i_` from a higher privilege mode
+is enabled as long as `clicintie[__i__]` is set (regardless of the setting
+of {status}.{ie} in the higher privilege modes).
+
+NOTE: This register bit is defined as WARL as unimplemented interrupts appear hardwired to zero.
+
+==== Interrupt Input Identification Number
+
+The 4096 CLIC interrupt inputs are given unique identification numbers
+with {mcause} Exception Code (`exccode`) values. 
+
+=== Indirect Access M-mode CLIC interrupt CSRs
+
+Access to CLIC registers `clicintattr[__i__]`, `clicintip[__i__]`, `clicintie[__i__]`, `cliciprio[__i__]`, and `clicideleg[__i__]`
+utilizes the Indirect CSR Access extension (Smcsrind/Sscsrind). Implementations may support
+another method to access these CSRs (e.g., via memory-mapped accesses) and any such a definition is outside the scope
+of the CLIC specification.
+
+If an interrupt _i_ is not present in the hardware, the corresponding CLIC register
+locations appear hardwired to zero.
+
+All CLIC registers are visible to M-mode.
+
+NOTE: Since accessing clic registers via indirect CSR access is not atomic, 
+indirect CSR access of these registers while same privilege mode {mstatus}.`mie` is enabled 
+requires `mireg` register state to be part of the interrupt handler's overall context state save/restore, 
+although this is expected to be an atypical need for most interrupt handlers.
+
+==== `clicintattr[__i__]`
+
+In this `miselect` offset range:
+
+* Each `mireg2` register controls the clic attribute setting of four interrupts
+
+[%autowidth]
+|===
+| `miselect` |  `mireg2` bits |  `mireg2` state              | description
+
+| 0x1000+i   |   7:0          | RW  `clicintattr[__i__*4+0]` |  setting for interrupt __i__*4+0

```

Comment:
the general strategy of register layout is different in AIA for RV64. When RV64, only every second register is used, and each CSR accesses 64b. This does increase efficiency for RV64

Reply:
and when I say RV64, i mean the the XLEN at the current mode is 64 :)